Core/AVIDump.cpp

```diff
@@ -0,0 +1,360 @@
+{
+	s_width = w;
+	s_height = h;
+	s_current_width = w;
+	s_current_height = h;
+
+	InitAVCodec();
+	bool success = CreateAVI();
+	if (!success)
+		CloseFile();
+	return success;
+}
+
+bool AVIDump::CreateAVI()
+{
+	AVCodec* codec = nullptr;
+
+	s_format_context = avformat_alloc_context();
+	std::stringstream s_file_index_str;
+	s_file_index_str << s_file_index;
+	snprintf(s_format_context->filename, sizeof(s_format_context->filename), "%s", (GetSysDirectory(DIRECTORY_VIDEO_DUMP) + "framedump" + s_file_index_str.str() + ".avi").c_str());
+	// Make sure that the path exists
+	if (!File::Exists(GetSysDirectory(DIRECTORY_VIDEO_DUMP)))
+		File::CreateDir(GetSysDirectory(DIRECTORY_VIDEO_DUMP));
+
+	if (File::Exists(s_format_context->filename))
+		File::Delete(s_format_context->filename);
+
+	if (!(s_format_context->oformat = av_guess_format("avi", nullptr, nullptr)) || !(s_stream = avformat_new_stream(s_format_context, codec)))
+	{
+		return false;
+	}
+
+	s_stream->codec->codec_id = g_Config.bUseFFV1 ? AV_CODEC_ID_FFV1 : s_format_context->oformat->video_codec;
+	if (!g_Config.bUseFFV1)
+		s_stream->codec->codec_tag = MKTAG('X', 'V', 'I', 'D');  // Force XVID FourCC for better compatibility
+	s_stream->codec->codec_type = AVMEDIA_TYPE_VIDEO;
+	s_stream->codec->bit_rate = 400000;
+	s_stream->codec->width = s_width;
+	s_stream->codec->height = s_height;
+	s_stream->codec->time_base.num = 1001;
+	s_stream->codec->time_base.den = 60000;
+	s_stream->codec->gop_size = 12;
+	s_stream->codec->pix_fmt = g_Config.bUseFFV1 ? AV_PIX_FMT_BGRA : AV_PIX_FMT_YUV420P;
+
+	if (!(codec = avcodec_find_encoder(s_stream->codec->codec_id)) || (avcodec_open2(s_stream->codec, codec, nullptr) < 0))
+	{
+		return false;
+	}
+
+	s_src_frame = av_frame_alloc();
+	s_scaled_frame = av_frame_alloc();
+
+	s_scaled_frame->format = s_stream->codec->pix_fmt;
+	s_scaled_frame->width = s_width;
+	s_scaled_frame->height = s_height;
+
+#if LIBAVCODEC_VERSION_MAJOR >= 55
+	if (av_frame_get_buffer(s_scaled_frame, 1))
+		return false;
+#else
+	if (avcodec_default_get_buffer(s_stream->codec, s_scaled_frame))
+		return false;
+#endif
+
+	NOTICE_LOG(G3D, "Opening file %s for dumping", s_format_context->filename);
+	if (avio_open(&s_format_context->pb, s_format_context->filename, AVIO_FLAG_WRITE) < 0 || avformat_write_header(s_format_context, nullptr))
+	{
+		WARN_LOG(G3D, "Could not open %s", s_format_context->filename);
+		return false;
+	}
+
+	return true;
+}
+
+static void PreparePacket(AVPacket* pkt)
+{
+	av_init_packet(pkt);
+	pkt->data = nullptr;
+	pkt->size = 0;
+	if (s_stream->codec->coded_frame->pts != AV_NOPTS_VALUE)
+	{
+		pkt->pts = av_rescale_q(s_stream->codec->coded_frame->pts,
+			s_stream->codec->time_base, s_stream->time_base);
+	}
+	if (s_stream->codec->coded_frame->key_frame)
+		pkt->flags |= AV_PKT_FLAG_KEY;
+	pkt->stream_index = s_stream->index;
+}
+
+static const u8 *ConvertBufferTo888RGB(const GPUDebugBuffer &buf, u8 *&temp, u32 &w, u32 &h) {
```

Comment:
If this is a straight copy, we should probably move it to somewhere common or expose it, rather than duplicating.

We may optimize this or fix bugs in it in the future.  A prime example: we may end up with a new debug buffer format from Vulkan.  Better to just have it one place.

-[Unknown]